packages/web/src/scss/components/Radio/_Radio.scss

```
@@ -70,54 +69,57 @@ $_field-name: 'Radio';
     @include form-fields-tools.label-required();
 }
 
-.Radio--disabled {
-    @include form-fields-tools.inline-field-root-disabled();
-}
+.Radio--item {
+    @include form-fields-tools.item();
 
-.Radio--disabled .Radio__label {
-    @include form-fields-tools.label-disabled();
+    display: flex;
 }
 
-.Radio--disabled .Radio__helperText {
-    @include form-fields-tools.helper-text-disabled();
+.Radio--item:is(.Radio--disabled, .Radio.is-disabled) {
+    @include form-fields-tools.item-disabled();
 }
 
-.Radio--disabled > .Radio__input,
-.Radio > .Radio__input:disabled {
-    border-color: form-fields-theme.$input-color-disabled;
-
-    &:checked {
-        border-color: form-fields-theme.$input-color-disabled;
-    }
+.Radio--item .Radio__label {
+    @include form-fields-tools.item-label();
 }
 
-.Radio--disabled > .Radio__input::before,
-.Radio > .Radio__input:disabled::before {
-    box-shadow: theme.$input-box-shadow-disabled; // 1.
+.Radio--item .Radio__helperText {
+    @include form-fields-tools.item-helper-text();
 }
 
-.Radio--item {
-    @include form-fields-tools.item();
+:is(.Radio--disabled, .Radio.is-disabled) {
+    @include form-fields-tools.inline-field-root-disabled();
+}
 
-    display: flex;
+:is(.Radio--disabled, .Radio.is-disabled) .Radio__label {
+    @include form-fields-tools.label-disabled();
 }
 
-.Radio--item.Radio--disabled {
-    @include form-fields-tools.item-disabled();
+:is(.Radio--disabled, .Radio.is-disabled) .Radio__helperText {
+    @include form-fields-tools.helper-text-disabled();
 }
 
-// stylelint-disable selector-max-class, selector-max-compound-selectors, selector-max-specificity -- We need to override the default styles for the label in Item.
-.Radio--item .Radio__input:checked ~ .Radio__label,
-.Radio--item .Radio__input:checked ~ .Radio__text > .Radio__label {
-    // stylelint-enable selector-max-class, selector-max-compound-selectors, selector-max-specificity
-    @include form-fields-tools.item-label-checked();
+:is(.Radio--disabled, .Radio.is-disabled) > .Radio__input,
+.Radio > .Radio__input:disabled {
+    border-color: form-fields-theme.$input-border-color-disabled;
+    background-color: form-fields-theme.$input-background-color-disabled;
+
+    &:checked {
+        border-color: form-fields-theme.$input-border-color-disabled;
+    }
+
+    &::before {
+        background-color: theme.$input-background-color-disabled;
+    }
 }
 
-.Radio--item .Radio__helperText {
-    @include form-fields-tools.item-helper-text();
+// stylelint-disable selector-max-specificity -- 1.
+.Radio--item:not(.Radio--disabled, .Radio.is-disabled) .Radio__input:is(:hover, :active):not(:checked) {
```

No need for the `.Radio > .Radio__input:disabled` variant? Not even like this?

```suggestion
.Radio--item:not(.Radio--disabled, .Radio.is-disabled) .Radio__input:is(:hover, :active):not(:checked, :disabled) {
```